src/nnet3/nnet-component-itf.cc

```diff
@@ -251,7 +251,6 @@ void NonlinearComponent::ZeroStats() {
 
 std::string NonlinearComponent::Info() const {
   std::stringstream stream;
-  KALDI_ASSERT(InputDim() == OutputDim());  // always the case
   stream << Type() << ", dim=" << InputDim();
```

Comment:
I had extended the NormalizeComponent::Info() to contain input-output dim if add-log-stddev is true.
I modified NonlinearComponent::Info() . Is this what you asked for?